ghost/core/core/server/services/comments/service.js

```diff
@@ -24,13 +26,25 @@ class CommentsService {
     }
 
     async sendNewCommentNotifications(comment) {
-        this.emails.notifyPostAuthors(comment);
+        await this.emails.notifyPostAuthors(comment);
 
         if (comment.get('parent_id')) {
-            this.emails.notifyParentCommentAuthor(comment);
+            await this.emails.notifyParentCommentAuthor(comment);
         }
     }
 
+    /**
+     * Dispatch an event that we created a new comments posted by given member
+     * @param {Object} member member model that posted the comment
+     */
+    dispatchCommentEvent(member) {
+        DomainEvents.dispatch(MemberCommentEvent.create({
+            memberId: member.id, 
+            memberLastSeenAt: member.get('last_seen_at'), 
```

Comment:
I know this is following the pattern we've used for the `MemberPageView` event - but this seems like we've coupled the event too tightly to the current event usage 🤔 

Perhaps A MemberCommentEvent should contain data more like:
```
{
  memberId,
  commentId,
  postId
}
```

And have the listener fetch the member - or use a service which handles checking whether or not we need to update 🤷 

Obviously that comes with a tradeoff of another database read, which if we're concerned about then perhaps we should just include the entire member in the payload? 

This isn't a huge deal, but it would be great if in the future when we want to perform some action when a member creates a comment, that we don't have to come here to edit the event, we'd have all the data we'd reasonably expect OR we'd have the pattern where we can fetch the data we need.

What do you think about this?

Reply:
Yes, I get what you mean :) Not sure if we should pass the models or the ids though. Passing only ids is definitely cleaner...

Reply:
Agreed ids is cleaner - we could also just pass JSON representation of the models 🤔

Reply:
OR - we merge as is, and add an issue somewhere to look at this in future, but not sure how likely it is we'll get to that

Reply:
Not gonna happen if we don't fix this now 😬

Reply:
I'm gonna go with the IDs I think.